api/base/serializers.py

```diff
@@ -280,6 +299,54 @@ def to_representation(self, data):
             self.child.to_representation(item, envelope=None) for item in data
         ]
 
+    # Overrides ListSerializer which doesn't support multiple update by default
+    def update(self, instance, validated_data):
+        if len(instance) != len(validated_data):
+            raise exceptions.NotFound()
+        id_lookup = self.child.fields['id'].source
+        instance_mapping = {getattr(item, id_lookup): item for item in instance}
+        data_mapping = {item.get('_id', None): item for item in validated_data}
+
+        ret = []
+
+        for resource_id, data in data_mapping.items():
+            resource = instance_mapping.get(resource_id, None)
+            ret.append(self.child.update(resource, data))
+
+        return ret
+
+    # overrides ListSerializer
+    def run_validation(self, data):
+        meta = getattr(self, 'Meta', None)
+        bulk_limit = getattr(meta, 'bulk_limit', REST_FRAMEWORK['DEFAULT_BULK_LIMIT'])
+
+        num_items = len(data)
+
+        if num_items > bulk_limit:
+            raise exceptions.ValidationError(
```

Comment:
Oh I remember why I was raising a non-field error: it was because I didn't want to return the entire large request under meta (if they had tried to bulk create 1000 items, say, and the limit was 10).  https://github.com/CenterForOpenScience/osf.io/pull/4353#discussion-diff-41794484R68  If it's a non-field error, I'm not returning the meta information.